Index: skalibs/AUTHORS
===================================================================
--- skalibs.orig/AUTHORS
+++ skalibs/AUTHORS
@@ -36,3 +36,4 @@ Thanks to:
   Johannes Nixdorf <mixi@shadowice.org>
   Casper Ti. Vector <caspervector@gmail.com>
   Guillermo <gdiazhartusch@gmail.com>
+  Alex Raschi <raschi.alex@gmail.com>
Index: skalibs/src/libstddjb/env_merge.c
===================================================================
--- skalibs.orig/src/libstddjb/env_merge.c
+++ skalibs/src/libstddjb/env_merge.c
@@ -16,7 +16,8 @@ size_t env_merge (char const **v, size_t
   {
     size_t split = str_chr(modifs + i, '=') ;
     size_t j = 0 ;
-    for (; j < vlen ; j++) if (!strncmp(modifs + i, v[j], split+1)) break ;
+    for (; j < vlen ; j++)
+      if (!strncmp(modifs + i, v[j], split) && (v[j][split] == '=')) break ;
     if (j < vlen) v[j] = v[--vlen] ;
     if (modifs[i + split]) v[vlen++] = modifs + i ;
   }
